echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/eventhandlers/TriggerEventHandler.java

```diff
@@ -58,19 +67,22 @@
    * Given a list of pipelines and an event, returns the pipelines that should be triggered
    * by the event
    * @param event The triggering event
-   * @param pipelines The pipelines to consider
+   * @param pipelineCache a source for pipelines and triggers to consider
    * @return The pipelines that should be triggered
    */
-  default List<Pipeline> getMatchingPipelines(T event, List<Pipeline> pipelines) {
-    if (isSuccessfulTriggerEvent(event)) {
-      return pipelines.stream()
-        .map(p -> withMatchingTrigger(event, p))
-        .filter(Optional::isPresent)
-        .map(Optional::get)
-        .collect(Collectors.toList());
-    } else {
+  default List<Pipeline> getMatchingPipelines(T event, PipelineCache pipelineCache) throws TimeoutException {
```

Comment:
this interface change (taking a `PipelineCache` instead of a `List<Pipeline>`) is what causes this PR to be so big, lots of call sites and tests had to be moved to the new interface. A bit unfortunate, but I think it is necessary: a `PipelineCache` can provide both the list of pipelines (for manual events) and the trigger index (for every other event handler)